platforms/core-runtime/build-state/src/main/java/org/gradle/internal/session/CrossBuildSessionState.java

```diff
@@ -90,68 +66,18 @@ public void close() {
     @SuppressWarnings("unused")
     private class Services {
 
-        private final StartParameter startParameter;
+        private final StartParameterInternal startParameter;
 
-        public Services(StartParameter startParameter) {
+        public Services(StartParameterInternal startParameter) {
             this.startParameter = startParameter;
         }
 
-        void configure(ServiceRegistration registration) {
```

Comment:
Moved some service registration back to core, where the services are implemented.